### DISCUSSION TITLE
Is there a way to kill/yank or otherwise access the info tide sends to the minibuffer?

### DISCUSSION THREAD

**Discussion by mwmx** [ananthakumaran/tide#431] [Q&A] [answered]
## Answer

For doc, you can run `M-x tide-documentation-at-point`, which should pop a buffer.  Be sure to have 
`(setq tide-always-show-documentation t)` somewhere.

For error messages, you can try `M-x flycheck-list-errors` or M-x `flycheck-copy-errors-as-kill`.

## Original question (plus a few minor edits)

It would be very helpful to be able to get access to the type information and error messages at point.

They display in the minibuffer but sometimes they're very long and don't fit.

And even when they do fit, I often find myself needing to copy this info around -- such as when a function expects a complex or unfamiliar type and I need to build up an argument for it myself.  Or, sometimes typescript error messages can be too long for me to be able to easily read; I'd love to be able to yank them into `*scratch*` and reformat them to something easier to read.

I've looked in `*Messages*`; they're not there.

**Comment by ananthakumaran** [comment, marked as answer]
For doc, you can run  `M-x tide-documentation-at-point`, which should pop a buffer. For error messages, you can try `M-x flycheck-list-errors` or `M-x flycheck-copy-errors-as-kill`

**Reply by mwmx** [reply]
I think I didn't explain what I am after very well.

```typescript
export type User = {
    id: number;
    name: string;
};

const u: User = { id: 1234, name: 'Alfalfa' };
//       ^
```

If I position the cursor just above the caret, I can see the definition of `User` in the minibuffer, but I can't figure out how to kill and then yank it.  `M-x tide-documentation-at-point` says "No documentation available."

Of course, in this example the definition is right there—so I'm not missing much.  But in less contrived situations you could be dealing with something in another file, possibly one deep inside `node_modules`.  You can always chase things down using `M-x tide-jump-to-definition` but usually what I want is right there the the minibuffer -- until you move point and then it's gone forever 😄 

If the output that goes to the minibuffer also went to `*Messages*` or a hypothetical `*tide-messages*` I think that would solve my problem nicely.

**Reply by mwmx** [reply]
Also, @ananthakumaran, thank you very much for the `flycheck` commands.  I didn't know about them.  I think they solve my other problem perfectly.  And thanks too for a very speedy response 😸

**Reply by ananthakumaran** [reply]
![Screenshot from 2021-07-25 10-03-18](https://user-images.githubusercontent.com/149238/126887863-1cb9b5b7-3df9-4f7e-a423-e11dd649dc3e.png)

I tried the above sample and could get the type info without any issue. (I don't get the "no documentation available" message)

**Reply by ananthakumaran** [reply]
I had forgotten about `tide-always-show-documentation`. Tide documentation at point by default doesn't show the buffer if there is no documentation string associated. This behavior can be overridden by the following configuration.

```
(setq tide-always-show-documentation t)
```

**Reply by mwmx** [reply]
With `tide-always-show-documentation` set to `t` it works.  Thank you very much.
